src/Symfony/Component/Validator/Constraints/DefaultPasswordStrengthEstimator.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Validator\Constraints;
+
+final class DefaultPasswordStrengthEstimator
+{
+    /**
+     * @var non-empty-array{score: int, minEntropy: int}
+     */
+    private array $criteria = [
+        ['score' => 4, 'minEntropy' => 120],
+        ['score' => 3, 'minEntropy' => 100],
+        ['score' => 2, 'minEntropy' => 80],
+        ['score' => 1, 'minEntropy' => 60],
+        ['score' => 0, 'minEntropy' => 0],
+    ];
+
+    /**
+     * @param string[] $restrictedData
+     *
+     * @return non-empty-array{score: int, restrictedData: string[]}
+     */
+    public function __invoke(#[\SensitiveParameter] string $password, array $restrictedData): array
+    {
+        $wordsFound = [];
+        if (\count($restrictedData) > 0) {
```

Comment:
I'm wondering if we should force each estimator to implement the handling of this restricted data. Wouldn't it be better to implement it in the ConstraintValidator, running this _before_ calling the estimator ? It would also make it a lot easier to describe the signature of the estimator callable, as this would receive a single argument being a string and would return a single argument being the score.

Reply:
I had in mind the behaviour of `zxcvbn`. I updated the PR to allow (basic) leet detection.
It could be part of another constraint if you prefer (e.g. `BannedWords`).